extensions-contrib/ganglia-emitter/src/main/java/org/apache/druid/emitter/ganglia/GangliaEmitter.java

```diff
@@ -0,0 +1,250 @@
+import org.apache.druid.java.util.common.concurrent.Execs;
+import org.apache.druid.java.util.common.logger.Logger;
+import org.apache.druid.java.util.emitter.core.Emitter;
+import org.apache.druid.java.util.emitter.core.Event;
+import org.apache.druid.java.util.emitter.service.ServiceMetricEvent;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.InetAddress;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.regex.Pattern;
+
+/**
+ * Created by yangxuan on 2018/9/6.
+ */
+public class GangliaEmitter implements Emitter
+{
+  private static final Logger log = new Logger(GangliaEmitter.class);
+  private static final char DRUID_METRIC_SEPARATOR = '/';
+  private static final Pattern STATSD_SEPARATOR = Pattern.compile("[:|]");
+  private static final Pattern BLANK = Pattern.compile("\\s+");
+
+  private final GangliaEmitterConfig config;
+  private final MetricRegistry registry;
+  private GMetric ganglia;
+  private final DimensionConverter converter;
+  private volatile ScheduledExecutorService scheduledExecutor;
+  private final ObjectMapper mapper;
+  private final LinkedBlockingQueue<GangliaEvent> eventsQueue;
+  private AtomicLong countLostEvents = new AtomicLong(0);
+  private final ScheduledExecutorService exec = Executors.newScheduledThreadPool(2, new ThreadFactoryBuilder()
+      .setDaemon(true)
+      .setNameFormat("GangliaEmitter-%s")
+      .build()); // Thread pool of two in order to schedule flush runnable
+
+  private Map<String, GangliaMetric> metricMap = new HashMap<>();
+
+  public GangliaEmitter(GangliaEmitterConfig config, ObjectMapper mapper)
+  {
+    this.config = config;
+    this.registry = new MetricRegistry();
+    this.converter = new DimensionConverter();
+    this.mapper = mapper;
+    this.eventsQueue = new LinkedBlockingQueue(config.getMaxQueueSize());
+  }
+
+  @Override
+  public void start()
+  {
+    try {
+      String spoof = InetAddress.getLocalHost().getHostAddress() + ":" + InetAddress.getLocalHost().getCanonicalHostName().split("\\.")[0];
+      log.info("Starting Ganglia Emitter");
+
+      this.ganglia = new GMetric(
+          config.getHostname(),
+          config.getPort(),
+          GMetric.UDPAddressingMode.UNICAST,
+          0,
+          true,
+          null,
+          spoof
+      );
+
+      //load config
+      scheduledExecutor = Execs.scheduledSingleThreaded("ganglia_metrics_load__scheduled_%d");
+      scheduledExecutor.scheduleAtFixedRate(
+          new Runnable()
+          {
+            @Override
+            public void run()
+            {
+              try {
+                metricMap = readMap(mapper, config.getDimensionMapPath());
```

Comment:
You don't strictly need to pass the `mapper` in here, it's already visible to the method from the instance field